db/migrations/20200203102132_create_bid_event_table.sql

```diff
@@ -0,0 +1,239 @@
+-- +goose Up
+CREATE TABLE maker.bid_event
+(
+    bid_id           NUMERIC     NOT NULL,
+    contract_address TEXT        NOT NULL,
+    act              api.bid_act NOT NULL,
+    lot              NUMERIC DEFAULT NULL,
+    bid_amount       NUMERIC DEFAULT NULL,
+    ilk_identifier   TEXT    DEFAULT NULL,
+    urn_identifier   TEXT    DEFAULT NULL,
+    block_height     BIGINT      NOT NULL
+);
+
+CREATE INDEX bid_event_index ON maker.bid_event (contract_address, bid_id);
+CREATE INDEX bid_event_urn_index ON maker.bid_event (ilk_identifier, urn_identifier);
+
+
+CREATE OR REPLACE FUNCTION maker.insert_bid_event(bid_id NUMERIC, address_id INTEGER, header_id INTEGER,
+                                                  act api.bid_act, lot NUMERIC, bid_amount NUMERIC) RETURNS VOID AS
+$$
+INSERT
+INTO maker.bid_event (bid_id, contract_address, act, lot, bid_amount, ilk_identifier, urn_identifier, block_height)
+VALUES (insert_bid_event.bid_id,
+        (SELECT address FROM public.addresses WHERE addresses.id = insert_bid_event.address_id),
+        insert_bid_event.act,
+        insert_bid_event.lot,
+        insert_bid_event.bid_amount,
+        (SELECT ilks.identifier
+         FROM maker.flip_ilk
+                  JOIN maker.ilks ON flip_ilk.ilk_id = ilks.id
+                  JOIN public.headers ON flip_ilk.header_id = headers.id
+         WHERE flip_ilk.address_id = insert_bid_event.address_id
+         ORDER BY headers.block_number DESC
+         LIMIT 1),
+        (SELECT usr
+         FROM maker.flip_bid_usr
+                  JOIN public.headers ON flip_bid_usr.header_id = headers.id
+         WHERE flip_bid_usr.bid_id = insert_bid_event.bid_id
+           AND flip_bid_usr.address_id = insert_bid_event.address_id
+         ORDER BY headers.block_number DESC
+         LIMIT 1),
+        (SELECT block_number FROM public.headers WHERE id = insert_bid_event.header_id))
+$$
+    LANGUAGE sql;
+
+CREATE OR REPLACE FUNCTION maker.delete_bid_event(bid_id NUMERIC, address_id INTEGER, header_id INTEGER,
+                                                  act api.bid_act, lot NUMERIC, bid_amount NUMERIC) RETURNS VOID AS
+$$
+DELETE
+FROM maker.bid_event
+    USING public.addresses, public.headers
+WHERE bid_event.contract_address = addresses.address
+  AND bid_event.block_height = headers.block_number
+  AND addresses.id = delete_bid_event.address_id
+  AND headers.id = delete_bid_event.header_id
+  AND bid_event.bid_id = delete_bid_event.bid_id
+  AND bid_event.act = delete_bid_event.act
+  AND bid_event.lot = delete_bid_event.lot
+  AND bid_event.bid_amount = delete_bid_event.bid_amount
```

Comment:
Looking at this again, I think this chunk is pretty silly 😅 would it make sense instead to include the `log_id` in this table, and cascade-delete when an `event_log` is deleted? Then we could omit this trigger altogether